projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -93,7 +97,17 @@ export class MapOlService {
   private hitTolerance = 0;
   /** 'olProjection' */
   public projectionChange = new Subject<olProjection>();
-  constructor() {
+  /**
+   * This object keeps track of currently bound angular-components that are being used as popups.
+   * We keep a reference to them here so that we can remove them again after they are no longer displayed.
+   */
+  private dynamicPopupComponents: {[key: string]: ComponentRef<any>} = {};
```

Comment:
Here you could use a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)
```ts
private dynamicPopupComponents: Map<string, ComponentRef<any>> = new Map();
```